.pipelines/templates/template-docker.yml

```diff
@@ -0,0 +1,147 @@
+parameters:
+  - name: jobName
+    type: string
+  - name: makeCommand
+    type: string
+  - name: repository
+    type: string
+  - name: acrFQDN
+    type: string
+  - name: imageRepoName
+    type: string
+  - name: pushLatest
+    type: boolean
+    default: false
+
+jobs:
+  - job: ${{ parameters.jobName }}
+    pool:
+      name: 1es-aro-ci-pool
+    variables:
+      ACR_FQDN: ${{ parameters.acrFQDN }}
+      REPOSITORY: ${{ parameters.repository }}
+      IMAGE_TAG: $(Build.BuildId)  # Include the unique BuildId or another identifier
+
+    steps:
+      - template: ../templates/template-checkout.yml
+
+      - task: AzureCLI@2
+        inputs:
+          azureSubscription: 'ado-pipeline-dev-image-push'  # Service connection name
+          scriptType: 'bash'
+          scriptLocation: 'inlineScript'
+          inlineScript: |
+            echo "Retrieving ACR token..."
+            REGISTRY_NAME=${ACR_FQDN%%.*}
+
+            if [ -z "$REGISTRY_NAME" ]; then
+              echo "Error: ACR_FQDN is not set correctly."
+              exit 1
+            fi
+            TOKEN=$(az acr login --name $REGISTRY_NAME --expose-token --output tsv --query accessToken)
+
+            if [ -z "$TOKEN" ]; then
+              echo "Error: Failed to retrieve ACR token."
+              exit 1
+            fi
+
+            echo "##vso[task.setvariable variable=ACR_TOKEN]$TOKEN"
+
+      - task: DockerInstaller@0
+        displayName: 'Install Docker'
+        inputs:
+          dockerVersion: '20.10.7'  # Adjust the version as needed
+
+      - script: |
```

Comment:
Alright - so we discovered due to cross-tenant auth issues, we can't use the `command: login` - so that's out, but we should be able to use current auth into ACR + `Docker@2` for a `buildAndPush` call to remove a bit of `sh` here, I think?